### PR TITLE
Added precheck in Lightning parallel import

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Added precheck content.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/6623
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 1]
```diff
@@ -112,6 +114,14 @@ target-table = "my_table"
 nohup ./tidb-lightning -config tidb-lightning.toml > nohup.out &
 ```
 
+在并行导入的场景下，iDB Lightning 在启动任务之后，会自动进行下列检查：
```
Comment (3pointer) [inline tidb-lightning/tidb-lightning-distributed-import.md:117]:
```suggestion
在并行导入的场景下，TiDB Lightning 在启动任务之后，会自动进行下列检查：
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 2]
```diff
@@ -18,11 +18,13 @@ TiDB Lightning 并行导入可以用于以下场景：
...
 > 并行导入只支持初始化 TiDB 的空表，不支持导入数据到已有业务写入的数据表，否则可能会导致数据不一致的情况。
 
-下图展示了并行导入分库分表的工作流程。
+下图展示了并行导入分库分表的工作流程。在该场景中，你可以使用多个 TiDB Lightning 实例导入 MySQL 的分表到下游的 TiDB集群。
+
 
 ![并行导入分库分表](/media/parallel-import-shard-tables.png)
 
-下图展示了并行导入单表的工作流程。
+下图展示了并行导入单表的工作流程。在该场景中，你可以使用多个 TiDB Lightning 实例，将单个表中的数据拆分后，并行导入到下游的 TiDB集群。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-distributed-import.md:26]:
```suggestion
下图展示了并行导入单表的工作流程。在该场景中，你可以使用多个 TiDB Lightning 实例，将单个表中的数据拆分后，并行导入到下游的 TiDB 集群。
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 3]
```diff
@@ -112,6 +114,15 @@ target-table = "my_table"
 nohup ./tidb-lightning -config tidb-lightning.toml > nohup.out &
 ```
 
+在并行导入的场景下，TiDB Lightning 在启动任务之后，会自动进行下列检查：
+
+- 检查本地盘空间以及 TiKV 集群是否有足够空间导入数据。检查时会对数据源进行采样，通过采样结果预估索引大小占比。由于估算中考虑了索引，因此可能会出现尽管数据源大小低于本地盘可用空间，但依然无法通过检测的情况。
+- 检查 TiKV 集群的 region 分布是否均匀，以及是否存在大量空 region，如果存在大量空 region，则无法执行导入。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-distributed-import.md:120]:
```suggestion
- 检查 TiKV 集群的 region 分布是否均匀，以及是否存在大量空 region，如果空 region 的数量大于 max(1000,  表的数量 * 3) ，即大于 1000 和 3 倍表数量二者中的最大者，则无法执行导入。
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+title: TiDB Lightning 执行前检查项
+summary: 本文档介绍了 TiDB Lightning 的检查项，确保 Lightning 检查顺利。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:3]:
```suggestion
summary: 本文档介绍了 TiDB Lightning 的检查项，确保 TiDB Lightning 检查顺利。
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+---
+
+# TiDB Lightning 执行前检查项
+
+在 v5.3 版本以及之后的 TiDB Lightning 中，增加了执行前检查配置的功能，默认开启，会进行一些磁盘空间，执行配置的常规检查，主要目的是确保后续的整个导入过程顺利。下面就检查具体内容进行说明:
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:8]:
```suggestion
从 TiDB 5.3.0 开始，TiDB Lightning 增加了执行前检查配置的功能，默认开启，会进行一些磁盘空间和执行配置的常规检查，主要目的是确保后续的整个导入过程顺利。

下表介绍了各检查项和详细解释。
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+|  检查项 | 解释 |
+|  ----  | ----  |
+| 集群版本/状态是否正常| 检查配置中集群是否可以连接，当后端模式是 Local 的时候，还会检查 TiKV/PD/TiFlash 版本是否支持 Local 导入模式。|
+| 是否有权限读取数据 | 检查当从云存储(S3)读取数据的时候，是否有对应的权限，确保不会因权限文档导致导入中断。|
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:13]:
```suggestion
| 是否有权限读取数据 | 检查当从云存储（Amazon S3）读取数据的时候，是否有对应的权限，确保不会因权限缺失导致导入中断。|
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+| 导入空间是否足够 | 检查 TiKV 集群是否有足够空间导入数据。检查时会对数据源进行采样，通过采样结果预估索引大小占比。由于估算中考虑了索引，因此可能会出现尽管数据源大小低于本地盘可用空间，但依然无法通过检测的情况。当后端是 Local 的时候，因为会在本地进行外部排序，所以还会检查本地存储是否足够 |
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:14]:
```suggestion
| 导入空间是否足够 | 检查 TiKV 集群是否有足够空间导入数据。检查时会对数据源进行采样，通过采样结果预估索引大小占比。由于估算中考虑了索引，因此可能会出现尽管数据源大小低于本地盘可用空间，但依然无法通过检测的情况。当后端是 Local 的时候，因为需要在本地进行外部排序，所以还会检查本地存储是否足够。
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+| region 分布状态 | 检查 TiKV 集群的 region 分布是否均匀，以及是否存在大量空 region，如果存在大量空 region，无法执行导入。 |
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:15]:
```suggestion
| region 分布状态 | 检查 TiKV 集群的 region 分布是否均匀，以及是否存在大量空 region，如果空 region 的数量大于 max(1000,  表的数量 * 3) ，即大于 “1000” 和 “3 倍表数量”二者中的最大者，无法执行导入。 |
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+| 数据文件是否有大 CSV 文件 | 当备份文件中出现大于 10G 的 CSV 文件且无法进行自动切分(StrictFormat=false)的时候，会导致导入性能下降，该检查的目的是提醒用户确保数据格式的情况下，开启自动切分 CSV 功能。 |
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:16]:
```suggestion
| 数据文件是否有大 CSV 文件 | 当备份文件中出现大于 10 GiB 的 CSV 文件且无法进行自动切分(StrictFormat=false)的时候，会导致导入性能下降。该检查的目的是提醒用户确保数据格式的情况下，开启自动切分 CSV 功能。 |
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+| 是否可以从断点恢复 | 该检查是确保断点恢复过程中，不会出现对源文件和数据库中 schema 进行修改，导致导入错误数据的情况。|
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:17]:
```suggestion
| 是否可以从断点恢复 | 该检查是确保断点恢复过程中，不会出现对源文件和数据库中 schema 进行修改，导致导入错误数据的情况。|
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+| 是否可以导入数据到已存在的数据表中| 当导入到已创建好的数据表中时，该检查尽可能的检查此次导入的源文件是否和已存在的数据表匹配。 检查列数是否匹配，如果源文件存在列名，则检查列名是否匹配。当源文件存在缺省列，则检查缺省列在数据表中是否存在 Default Value，如果存在，则检查通过 ｜
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:18]:
```suggestion
| 是否可以导入数据到已存在的数据表中| 当导入到已创建好的数据表中时，该检查尽可能的检查此次导入的源文件是否和已存在的数据表匹配。检查列数是否匹配，如果源文件存在列名，则检查列名是否匹配。当源文件存在缺省列，则检查缺省列在数据表中是否存在 Default Value，如果存在，则检查通过。｜
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 12]
```diff
@@ -18,11 +18,13 @@ TiDB Lightning 并行导入可以用于以下场景：
 >
 > 并行导入只支持初始化 TiDB 的空表，不支持导入数据到已有业务写入的数据表，否则可能会导致数据不一致的情况。
 
-下图展示了并行导入分库分表的工作流程。
+下图展示了并行导入分库分表的工作流程。在该场景中，你可以使用多个 TiDB Lightning 实例导入 MySQL 的分表到下游的 TiDB集群。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-distributed-import.md:21]:
```suggestion
下图展示了并行导入分库分表的工作流程。在该场景中，你可以使用多个 TiDB Lightning 实例导入 MySQL 的分表到下游的 TiDB 集群。
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 13]
```diff
@@ -112,6 +114,15 @@ target-table = "my_table"
...
 ```
 
+在并行导入的场景下，TiDB Lightning 在启动任务之后，会自动进行下列检查：
+
+- 检查本地盘空间以及 TiKV 集群是否有足够空间导入数据。检查时会对数据源进行采样，通过采样结果预估索引大小占比。由于估算中考虑了索引，因此可能会出现尽管数据源大小低于本地盘可用空间，但依然无法通过检测的情况。
+- 检查 TiKV 集群的 region 分布是否均匀，以及是否存在大量空 region，如果空 region 的数量大于 max(1000,  表的数量 * 3) ，即大于 “1000” 和 “3 倍表数量”二者中的最大者，则无法执行导入。
+- 检查数据源导入数据是否有序，并且根据检查结果自动调整 `mydumper.batch-size` 的大小。因此 `mydumper.batch-size` 配置不再对用户开放。
+
+你也可以通过 `lightning.check-requirements` 配置来关闭检查，执行强制导入。更多详细检查内容，可以查看[Lightning 执行前检查项](/tidb-lightning/tidb-lightning-prechecks.md)
+
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-distributed-import.md:124]:
```suggestion
```

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+
+从 TiDB 5.3.0 开始，TiDB Lightning 增加了执行前检查配置的功能。默认开启。该功能会自动进行一些磁盘空间和执行配置的常规检查，主要目的是确保后续的整个导入过程顺利。
+
+下表介绍了各检查项和详细解释。
+
+|  检查项 | 解释 |
+|  ----  | ----  |
+| 集群版本/状态是否正常| 检查配置中集群是否可以连接，当后端模式是 Local 的时候，还会检查 TiKV/PD/TiFlash 版本是否支持 Local 导入模式。|
+| 是否有权限读取数据 | 检查当从云存储（Amazon S3）读取数据的时候，是否有对应的权限，确保不会因权限缺失导致导入中断。|
+| 导入空间是否足够 | 检查 TiKV 集群是否有足够空间导入数据。检查时会对数据源进行采样，通过采样结果预估索引大小占比。由于估算中考虑了索引，因此可能会出现尽管数据源大小低于本地盘可用空间，但依然无法通过检测的情况。当后端是 Local 的时候，因为需要在本地进行外部排序，所以还会检查本地存储是否足够。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-prechecks.md:16]:
```suggestion
| 导入空间是否足够 | 检查 TiKV 集群是否有足够空间导入数据。检查时会对数据源进行采样，通过采样结果预估索引大小占比。由于估算中考虑了索引，因此可能会出现尽管数据源大小低于本地盘可用空间，但依然无法通过检测的情况。当后端是 Local 的时候，因为需要在本地进行外部排序，所以还会检查本地存储是否足够。| 
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 15]
```diff
@@ -112,6 +112,14 @@ target-table = "my_table"
 nohup ./tidb-lightning -config tidb-lightning.toml > nohup.out &
 ```
 
+在并行导入的场景下，TiDB Lightning 在启动任务之后，会自动进行下列检查：
+
+- 检查本地盘空间以及 TiKV 集群是否有足够空间导入数据。检查时会对数据源进行采样，通过采样结果预估索引大小占比。由于估算中考虑了索引，因此可能会出现尽管数据源大小低于本地盘可用空间，但依然无法通过检测的情况。
+- 检查 TiKV 集群的 region 分布是否均匀，以及是否存在大量空 region，如果空 region 的数量大于 max(1000,  表的数量 * 3) ，即大于 “1000” 和 “3 倍表数量”二者中的最大者，则无法执行导入。
+- 检查数据源导入数据是否有序，并且根据检查结果自动调整 `mydumper.batch-size` 的大小。因此 `mydumper.batch-size` 配置不再对用户开放。
+
+你也可以通过 `lightning.check-requirements` 配置来关闭检查，执行强制导入。更多详细检查内容，可以查看[Lightning 执行前检查项](/tidb-lightning/tidb-lightning-prechecks.md)
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-distributed-import.md:121]:
```suggestion
你也可以通过 `lightning.check-requirements` 配置来关闭检查，执行强制导入。更多详细检查内容，可以查看 [Lightning 执行前检查项](/tidb-lightning/tidb-lightning-prechecks.md)。
```
强制导入是否不推荐用户去做？强制导入有什么可能的后果，需要提前告知用户吗?

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 16]
```diff
@@ -0,0 +1,20 @@
+---
+title: TiDB Lightning 执行任务前检查项
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-prechecks.md:2]:
```suggestion
title: TiDB Lightning 执行任务前的检查项
```

File: tidb-lightning/tidb-lightning-distributed-import.md (hunk 15 above)
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-distributed-import.md:121]:
```suggestion
你也可以通过 `lightning.check-requirements` 配置来关闭检查，执行强制导入。更多详细检查内容，可以查看 [Lightning 执行前检查项](/tidb-lightning/tidb-lightning-prechecks.md)。
```
这个推荐用户关闭吗？如果关闭，可能导致的后果，需要提前告知用户吗？

File: tidb-lightning/tidb-lightning-prechecks.md [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+
+|  检查项 | 解释 |
+|  ----  | ----  |
+| 集群版本/状态是否正常| 检查配置中集群是否可以连接，当后端模式是 Local 的时候，还会检查 TiKV/PD/TiFlash 版本是否支持 Local 导入模式。|
+| 是否有权限读取数据 | 检查当从云存储（Amazon S3）读取数据的时候，是否有对应的权限，确保不会因权限缺失导致导入中断。|
+| 导入空间是否足够 | 检查 TiKV 集群是否有足够空间导入数据。检查时会对数据源进行采样，通过采样结果预估索引大小占比。由于估算中考虑了索引，因此可能会出现尽管数据源大小低于本地盘可用空间，但依然无法通过检测的情况。当后端是 Local 的时候，因为需要在本地进行外部排序，所以还会检查本地存储是否足够。| 
+| region 分布状态 | 检查 TiKV 集群的 region 分布是否均匀，以及是否存在大量空 region，如果空 region 的数量大于 max(1000,  表的数量 * 3) ，即大于 “1000” 和 “3 倍表数量”二者中的最大者，无法执行导入。 |
+| 数据文件是否有大 CSV 文件 | 当备份文件中出现大于 10 GiB 的 CSV 文件且无法进行自动切分(StrictFormat=false)的时候，会导致导入性能下降。该检查的目的是提醒用户确保数据格式的情况下，开启自动切分 CSV 功能。 |
+| 是否可以从断点恢复 | 该检查是确保断点恢复过程中，不会出现对源文件和数据库中 schema 进行修改，导致导入错误数据的情况。|
+| 是否可以导入数据到已存在的数据表中| 当导入到已创建好的数据表中时，该检查尽可能的检查此次导入的源文件是否和已存在的数据表匹配。检查列数是否匹配，如果源文件存在列名，则检查列名是否匹配。当源文件存在缺省列，则检查缺省列在数据表中是否存在 Default Value，如果存在，则检查通过。｜
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-prechecks.md:20]:
```suggestion
| 是否可以导入数据到已存在的数据表中| 当导入到已创建好的数据表中时，该检查尽可能的检查此次导入的源文件是否和已存在的数据表匹配。检查列数是否匹配，如果源文件存在列名，则检查列名是否匹配。当源文件存在缺省列，则检查缺省列在数据表中是否存在 Default Value，如果存在，则检查通过。|
```

**Review by Liuxiaozhen12 — APPROVED** [review]

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
@hfxsd The `translation/done` label means that a corresponding docs PR has been created for this PR. Please link this PR to the docs PR in the description part. Thanks!

**Pull request merged**